Drop validation and references to U2F Facets

codingllama opened the pull request:
U2F facets don't serve a purpose since Webauthn became the de-facto MFA mechanism. This should make it easier for users to write v10-style authentication configs.

The remaining U2F fields are still relevant for when we derive Webauthn settings from U2F, so those are kept (and validated).

Note that we still keep the fileconf field, since we don't want configs to needlessly break during version upgrades.

#10375

codingllama committed:
Remove Facets from U2F proto

codingllama committed:
Update generated protos

codingllama committed:
Remove references to U2F.Facets

zmb3 commented on lib/config/fileconf.go:
```diff
@@ -854,7 +854,9 @@ func (a *AuthenticationConfig) Parse() (types.AuthPreference, error) {
+	AppID string `yaml:"app_id"`
+	// Facets kept only to avoid breakages during Teleport updates.
+	// Webauthn is now used instead of U2F.
 	Facets               []string `yaml:"facets"`
```
Should we maybe log a warning and mark this for deletion in Teleport 11 or 12 (by which time nobody should be using U2F anymore)?

codingllama committed:
Reinstate proto Facets field

U2F gets marshaled as JSON to storage, so we better keep the field to
avoid issue with resource configs.

greedy52 commented on lib/config/fileconf.go:
```diff
@@ -854,7 +854,9 @@ func (a *AuthenticationConfig) Parse() (types.AuthPreference, error) {
 type UniversalSecondFactor struct {
-	AppID                string   `yaml:"app_id"`
+	AppID string `yaml:"app_id"`
+	// Facets kept only to avoid breakages during Teleport updates.
```
nit
```suggestion
	// Deprecated: Facets kept only to avoid breakages during Teleport updates.
```

codingllama replied:
Done (comment is slightly different, but added "Deprecated: " at the start).

codingllama committed:
Add deprecation notice on comment

codingllama commented:
Friendly ping @zmb3 ?
